src/index.js

```diff
@@ -38,12 +38,29 @@ function cachedIsFile (file, cb) {
 	isFileCache[file].then(contents => cb(null, contents), cb);
 }
 
+function deprecatedMainField (options, option, mainFields, field = option) {
+	if (option in options) {
+		CONSOLE_WARN(`node-resolve: setting options.${option} is deprecated, please override options.mainFields instead`);
+		if (options[option] === false) {
+			return mainFields.filter(mainField => mainField === field);
+		} else if (options[option] === true && mainFields.indexOf(field) === -1) {
+			return [field].concat(mainFields);
```

Comment:
why do you prepend the field and not append it? Doesn't this line mean that - if a user sets browser, module, jsnext and main to true, she will receive an array `[main, jsnext, module, browser]`? And doesn't that in turn mean that the precedence order of old lines 140-146 is now reversed? Or did you reverse that order on purpose?

Reply:
thanks for mainField support. and i'v fixed the fields order issues at <https://github.com/fedorio/rollup-plugin-node-resolve/blame/b63feb54ba226681fecbf690936b10f0dcb521f5/src/index.js#L89-L99>

```js
	let mainFields = options.mainFields || [];
	const fields = { 'browser': 0, 'module': 0, 'jsnext': 'jsnext:main', 'main': 0 };
	const keys = Object.keys(options).filter(k => fields.hasOwnProperty(k));


	// build mainFields by plugins options (keep options order)
	mainFields = keys.reduce((list, k) => deprecatedMainField(options, k, list, fields[k] || k), mainFields);


	// set defaults mainFields if none options
	[ 'module', 'main' ].forEach(k => {
		if (!mainFields.includes(k) && options[k] !== false) mainFields.push(k);
	});
```